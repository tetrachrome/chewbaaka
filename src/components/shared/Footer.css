/**
 * Footer.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 19, 2020
 * Updated  : Aug 19, 2020
 */

p.FooterColumnHeading {
  font-size: 18px;
  color: white;
}

a.CCFLink {
  color: rgb(255,182,31);
}

a.CCFLink:hover {
  color: rgb(255,149,0);
}

@media (max-width: 415px) {
  .FooterNav a {
    display: inline-block;
    font-size: 17px;
    margin: 1rem auto;
    padding: 20px;
  }
}
