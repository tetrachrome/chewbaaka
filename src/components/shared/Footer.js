/**
 * Footer.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 26, 2020
 */

import React from 'react'

import {
  Container,
  Grid,
  Header,
  Image,
  List,
  Segment
} from 'semantic-ui-react'

import 'semantic-ui-css/semantic.min.css'

import { kStringConstantCheetahConservationFund } from './constants'

import logo from './assets/cheetah-conservation-fund-logo.jpg'

export default function Footer() {
  return (
    <Segment inverted style={{ padding: "5em 0em" }} data-testid="FooterComponentTestId">
      <Container>
        <Grid divided inverted stackable>
          <Grid.Row>
            <Grid.Column width={3}>
              <Image
                size="small"
                src={logo}
              />
            </Grid.Column>
            <Grid.Column width={6}>
              <Header inverted as="h4" content="Resources" />
              <List link inverted>
                <List.Item as="a" target="_blank" href="https://cheetah.org/about/what-we-do/">About CCF</List.Item>
                <List.Item as="a" target="_blank" href="https://animalfactguide.com/animal-facts/cheetah/">Cheetah Fun Facts</List.Item>
                <List.Item as="a" target="_blank" href="https://wildnet.org/wildlife-programs/cheetah-namibia/">Wildlife Conservation Network & CCF</List.Item>
                <List.Item as="a" target="_blank" href="https://www.iucnredlist.org/species/219/50649567">IUCN Red List - Cheetah</List.Item>
              </List>
            </Grid.Column>
            <Grid.Column width={7}>
              <Header as="h4" inverted>
                {kStringConstantCheetahConservationFund}
              </Header>
              <p data-testid="footerSectionCCF">
                <a href="https://cheetah.org/">{kStringConstantCheetahConservationFund}</a> is a 501(c)(3) nonprofit organization,
                dedicated to saving the cheetahs in the wild and changing
                the world to be a better place.
              </p>
              <p>
                {`This site is not an affiliation of ${kStringConstantCheetahConservationFund}.`}
              </p>
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Container>
    </Segment>
  );
}
