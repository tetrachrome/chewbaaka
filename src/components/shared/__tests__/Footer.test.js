/**
 * Footer.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 02, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import Footer from '../Footer';

test('renders global footer', () => {
  render(<Footer />);

  // Tests section title with 'Resources' is present.
  const sectionTitleResources = screen.getByText(/Resources/i);
  expect(sectionTitleResources).toBeInTheDocument();

  // Tests section title with 'CCF' is present.
  const sectionElementCCF = screen.getByTestId("footerSectionCCF");
  expect(sectionElementCCF).toBeInTheDocument();

  // Tests section text.
  const sectionElementCCFTextContent = sectionElementCCF.textContent;
  expect(sectionElementCCFTextContent.includes("Cheetah Conservation Fund")).toBeTruthy();
  expect(sectionElementCCFTextContent.includes("501(c)(3)")).toBeTruthy();

  // Tests disclaimer text to be present.
  const disclaimerItem = screen.getByText(/This site is not an affiliation of Cheetah Conservation Fund/i);
  expect(disclaimerItem).toBeInTheDocument();
});
