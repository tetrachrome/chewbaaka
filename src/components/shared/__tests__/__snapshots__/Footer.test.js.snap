// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component snapshot 1`] = `
<div
  className="ui inverted segment"
  data-testid="FooterComponentTestId"
  style={
    Object {
      "padding": "5em 0em",
    }
  }
>
  <div
    className="ui container"
  >
    <div
      className="ui inverted stackable divided grid"
    >
      <div
        className="row"
      >
        <div
          className="three wide column"
        >
          <img
            className="ui small image"
            src="cheetah-conservation-fund-logo.jpg"
          />
        </div>
        <div
          className="six wide column"
        >
          <h4
            className="ui inverted header"
          >
            Resources
          </h4>
          <div
            className="ui inverted link list"
            role="list"
          >
            <a
              className="item"
              href="https://cheetah.org/about/what-we-do/"
              onClick={[Function]}
              role="listitem"
              target="_blank"
            >
              About CCF
            </a>
            <a
              className="item"
              href="https://animalfactguide.com/animal-facts/cheetah/"
              onClick={[Function]}
              role="listitem"
              target="_blank"
            >
              Cheetah Fun Facts
            </a>
            <a
              className="item"
              href="https://wildnet.org/wildlife-programs/cheetah-namibia/"
              onClick={[Function]}
              role="listitem"
              target="_blank"
            >
              Wildlife Conservation Network & CCF
            </a>
            <a
              className="item"
              href="https://www.iucnredlist.org/species/219/50649567"
              onClick={[Function]}
              role="listitem"
              target="_blank"
            >
              IUCN Red List - Cheetah
            </a>
          </div>
        </div>
        <div
          className="seven wide column"
        >
          <h4
            className="ui inverted header"
          >
            Cheetah Conservation Fund
          </h4>
          <p
            data-testid="footerSectionCCF"
          >
            <a
              href="https://cheetah.org/"
              rel=" noopener noreferrer"
              target="_blank"
            >
              Cheetah Conservation Fund
            </a>
             is a 501(c)(3) nonprofit organization, dedicated to saving the cheetah in the wild and changing the world to be a better place.
          </p>
          <p>
            This site is not an affiliation of Cheetah Conservation Fund.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
